test/pickfirst_test.go

```diff
@@ -382,6 +383,8 @@ func (s) TestPickFirst_StickyTransientFailure(t *testing.T) {
 }
 
 func (s) TestPickFirst_ShuffleAddressList(t *testing.T) {
```

Comment:
I'd rather not be detailed and explain the whole test setup and everything in the comment.  The name is very self-explanatory, but I added a short comment.

Reply:
That's not what I mean. I just feel like it should be briefly explain stuff like configures pick_first as top level lb of the channel, sends it address list before knob and connection isn't random, after knob it is random (with our determinisitic randomness algorithm plumbed). See https://github.com/grpc/grpc-go/blob/8d2f69a2ba8ab45dbbedb1e9dd93654332544db4/test/pickfirst_test.go#L239 for an example of what I'm going for.

Reply:
This comment is unnecessarily detailed and a recipe for the test & comment diverging when someone decides to change the test to add an extra check or do things in a different order because of a race, etc.  I'd rather not over-document our tests, and would definitely not require detailed documentation anywhere.  It's actually pretty rare to have _any_ comments at the top of a test.

E.g. https://cs.opensource.google/go/go/+/refs/tags/go1.20.4:src/net/http/client_test.go;l=550